spring-components/rdf4j-spring/src/main/java/org/eclipse/rdf4j/spring/support/ShaclSailValidationReportHelper.java

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.spring.support;
+
+import java.io.StringWriter;
+import java.lang.invoke.MethodHandles;
+
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.Rio;
+import org.eclipse.rdf4j.sail.shacl.ShaclSailValidationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @since 4.0.0
+ * @author Florian Kleedorfer
+ */
+public class ShaclSailValidationReportHelper {
+	private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+	public static void logValidationReportFromThrowableCause(Throwable t) {
+		Throwable thowable;
+		for (thowable = t; thowable != null; thowable = thowable.getCause()) {
+			if (thowable instanceof ShaclSailValidationException) {
+				break;
+			}
+		}
+		if (thowable != null && thowable instanceof ShaclSailValidationException) {
+			Model model = ((ShaclSailValidationException) thowable).validationReportAsModel();
+			StringWriter w = new StringWriter();
+			Rio.write(model, System.err, RDFFormat.TURTLE);
+		}
+	}
+}
```

Comment:
Could you try to check for ValidationException instead?

I could also recommend pretty printing and blank node inlining.

Reply:
More generally: should this helper code live here? It does not look like something specific to spring, at first glance. Though the same is true for SHACL's `ValidationReport` class itself - perhaps this is a good opportunity to find a good spot for both these things in one of the common-packages?

Reply:
> More generally: should this helper code live here? It does not look like something specific to spring, at first glance. Though the same is true for SHACL's `ValidationReport` class itself - perhaps this is a good opportunity to find a good spot for both these things in one of the common-packages?

If not here, then probably the `rdf4j-shacl` module, right? You won't need this class unless you are (or might be) using shacl. In that case, I'd put it in `src/main/java/org/eclipse/rdf4j/sail/shacl`. Ok?

> Could you try to check for ValidationException instead?

yes. good point.
 
> I could also recommend pretty printing and blank node inlining.

Good points, too. I would have liked these features when working on our shacl shapes. Following blank node ids around is tedious ;-)

Reply:
addressed these in the latest commits to the PR